src/config-file.c

```diff
@@ -61,7 +61,7 @@ static gboolean get_key_string(GKeyFile *key_file, const gchar* group, const gch
 static gboolean get_key_bool(GKeyFile *key_file, const gchar* group, const gchar* key, gboolean* value, const gboolean default_value, GError **error)
 {
         g_autofree gchar *val = NULL;
-        val = g_key_file_get_string(key_file, group, key, NULL);
+        val = g_key_file_get_string(key_file, group, key, error);
```

Comment:
This won't help. It sets error but the line below ensures that the function unconditionally returns `TRUE`, anyway.
Thus there is no chance that this error would be handled.